File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Projections/OperationArrayExpansionPersistence.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder
{
    using Microsoft.CommonDataModel.ObjectModel.Cdm;
    using Microsoft.CommonDataModel.ObjectModel.Enums;
    using Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types;
    using Microsoft.CommonDataModel.ObjectModel.Utilities;
    using Newtonsoft.Json.Linq;

    /// <summary>
    /// Operation ArrayExpansion persistence
    /// </summary>
    public class OperationArrayExpansionPersistence
    {
        public static CdmOperationArrayExpansion FromData(CdmCorpusContext ctx, JToken obj)
        {
            if (obj == null)
            {
                return null;
            }

            CdmOperationArrayExpansion arrayExpansionOp = OperationBasePersistence.FromData<CdmOperationArrayExpansion>(ctx, CdmObjectType.OperationArrayExpansionDef, obj);
            arrayExpansionOp.StartOrdinal = (int?)obj["startOrdinal"];
            arrayExpansionOp.EndOrdinal = (int?)obj["endOrdinal"];

            return arrayExpansionOp;
        }

        public static OperationArrayExpansion ToData(CdmOperationArrayExpansion instance, ResolveOptions resOpt, CopyOptions options)
        {
            if (instance == null)
            {
                return null;
            }

            OperationArrayExpansion obj = OperationBasePersistence.ToData<OperationArrayExpansion>(instance, resOpt, options);
            obj.StartOrdinal = instance.StartOrdinal;
            obj.EndOrdinal = instance.EndOrdinal;

            return obj;
        }
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/projections/OperationArrayExpansionPersistence.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.projections;

import com.fasterxml.jackson.databind.JsonNode;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmCorpusContext;
import com.microsoft.commondatamodel.objectmodel.cdm.projections.CdmOperationArrayExpansion;
import com.microsoft.commondatamodel.objectmodel.cdm.projections.OperationTypeConvertor;
import com.microsoft.commondatamodel.objectmodel.enums.CdmObjectType;
import com.microsoft.commondatamodel.objectmodel.enums.CdmOperationType;
import com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types.projections.OperationArrayExpansion;
import com.microsoft.commondatamodel.objectmodel.utilities.CopyOptions;
import com.microsoft.commondatamodel.objectmodel.utilities.ResolveOptions;
import com.microsoft.commondatamodel.objectmodel.utilities.StringUtils;
import com.microsoft.commondatamodel.objectmodel.utilities.logger.Logger;

/**
 * Operation ArrayExpansion persistence
 */
public class OperationArrayExpansionPersistence {
    public static CdmOperationArrayExpansion fromData(final CdmCorpusContext ctx, final JsonNode obj) {
        if (obj == null) {
            return null;
        }

        CdmOperationArrayExpansion arrayExpansionOp = OperationBasePersistence.fromData(ctx, CdmObjectType.OperationArrayExpansionDef, obj);
        arrayExpansionOp.setStartOrdinal(obj.get("startOrdinal") == null ? null : obj.get("startOrdinal").asInt());
        arrayExpansionOp.setEndOrdinal(obj.get("endOrdinal") == null ? null : obj.get("endOrdinal").asInt());

        return arrayExpansionOp;
    }

    public static OperationArrayExpansion toData(final CdmOperationArrayExpansion instance, ResolveOptions resOpt, CopyOptions options) {
        if (instance == null) {
            return null;
        }

        OperationArrayExpansion obj = OperationBasePersistence.toData(instance, resOpt, options);
        obj.setStartOrdinal(instance.getStartOrdinal());
        obj.setEndOrdinal(instance.getEndOrdinal());

        return obj;
    }
}
